actionpack/lib/action_dispatch/testing/request_encoder.rb

```diff
@@ -50,6 +52,7 @@ def self.register_encoder(mime_name, param_encoder: nil, response_parser: nil)
       @encoders[mime_name] = new(mime_name, param_encoder, response_parser)
     end
 
+    register_encoder :html, response_parser: -> body { Nokogiri::HTML5.parse(body) }
```

Comment:
Just a note that the `Nokogiri::HTML5` module/namespace will not exist for Nokogiri `< 1.12.0` or for any JRuby user (see https://github.com/sparklemotion/nokogiri/issues/2227).

I think we should take care to gracefully fall back to `Nokogiri::HTML.parse(body)` unless `defined?(::Nokogiri::HTML5)`